lib/proxy/server.go

```diff
@@ -147,7 +148,11 @@ func NewServer(config ServerConfig) (*Server, error) {
 
 // Serve starts the proxy server.
 func (s *Server) Serve() error {
-	if err := s.server.Serve(s.config.Listener); err != nil && err != grpc.ErrServerStopped {
+	if err := s.server.Serve(s.config.Listener); err != nil {
+		if errors.Is(err, grpc.ErrServerStopped) ||
+			errors.Is(err, net.ErrClosed) {
```

Comment:
Do we want to use `utils.IsUseOfClosedNetworkError(err)` here, for consistency with the rest of the codebase?